### PR TITLE
FEAT: Forgot Password Endpoint CU-2rjdgbd

### PR DESCRIPTION
**Pull request by drewlacob**
- add forgot password endpoint which creates a temporary password, sets it in the db, and emails it to the given email
- setup nodemailer and create mailer util to handle mail requests
- set up SiB accounts, hooked in through .env.local and then config
- update api

### CODE REVIEW

**Comment by freedom-rains** [comment]
Task linked: [CU-2rjdgbd BE - Forgot Password endpoint ](https://app.clickup.com/t/2rjdgbd)

**Review by RealDavidH — COMMENTED** [review]
How do we only give the temp password to the user who needs it? What if I put in your email and get a temp password for your account?

File: server/utils/mailer.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      service: 'SendinBlue',
+      port: 587,
+      auth: {
+        user: emailUsername,
+        pass: emailSMTPPassword,
+      },
+    });
+
+    await transporter.sendMail({
+      from: 'freedomtvplus@gmail.com',
```
Comment (freedom-rains) [inline server/utils/mailer.js:17]: Change to be `info@freedomrains.com`

**Review by freedom-rains — CHANGES_REQUESTED** [review]

**Comment by freedom-rains** [comment]
> How do we only give the temp password to the user who needs it? What if I put in your email and get a temp password for your account?

@RealDavidH say I type in your email address in the “Forgot Password” section of the website. Even if you weren’t the one to do it, you will still get the email letting you know your password has been temporarily reset. Then you would just have to use the temporary password to login and then change you password. I (as the person who typed in your email) should never get the email that has your new temporary password.

**Review by freedom-rains — APPROVED** [review]

**Comment by AlamTahmidul** [comment]
Shouldn't we return the response status codes instead of ending them to make sure that at least something gets sent back to the client?

**Review by mclaurind — APPROVED** [review]

**Comment by drewlacob** [comment]
> Shouldn't we return the response status codes instead of ending them to make sure that at least something gets sent back to the client?

Can you mark the lines that you are talking about? Do you mean in the mailer util? In the user controller it handles if it was sent or not and returns a status based on that.

**Pull request merged**